### PR TITLE
feat(cloudflare-module): support `cloudflare:email` and `cloudflare:queue` hooks

### PR DESCRIPTION
**Pull request by dalbodeule**
### 🔗 Linked issue

[https://github.com/unjs/nitro/issues/2480](https://github.com/unjs/nitro/issues/2480)

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The existing Cloudflare-module preset did not have a function that could handle Email.

To solve this, we worked on email handlers.
Similar work was performed with the existing task handler. I'm not sure if it's right to do this.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/presets/cloudflare/runtime/cloudflare-module.ts [hunk 1]
```diff
@@ -96,6 +96,26 @@ export default {
...
     }
   },
+  email(event: any, env: CFModuleEnv, context: ExecutionContext) {
+    if (import.meta._tasks) {
+      (globalThis as any).__env__ = env;
+
+      const domain = (event.to as string).split("@")[1];
+
+      context.waitUntil(
+        runEmailTask(domain, {
```
Comment (pi0) [inline src/presets/cloudflare/runtime/cloudflare-module.ts:106]: Here, we shall simply call `event.waitUntil(nitroApp.hooks.callHook('cloudflare:email', event, { env, context })`

Comment (dalbodeule) [reply src/presets/cloudflare/runtime/cloudflare-module.ts:106]: Is it right "cloudflare:email" hooks on here?

https://github.com/unjs/nitro/blob/40a0007a65440187d236efdeb1403c8c301bafe1/src/runtime/types.ts#L24

**Comment by dalbodeule** [comment]
I did what was asked once. Please confirm if this is what you think.

If you make it like this

```ts
defineNuxtPlugin((nitroApp) => {
 nitroApp.hooks.hook("cloudflare:email", handler);
})
```

I want to check whether a pattern is used.

If this is the pattern. I think it would be okay to create a cloudflare:queue handler in the same way.

**Comment by dalbodeule** [comment]
As a new version of nitro was released, Merge, email handler, and queue handler were added to suit the new version. Request you to check.

**Review by pi0 — APPROVED** [review]
Thanks ❤️

**Pull request merged**